Add timeline for future aside component

Cosmin-Daniel opened the pull request:
The timeline component is added as definition list on the Research Article page to be used inside the new aside component.

viktoria-mahmud commented on assets/sass/patterns/atoms/definition-list.scss:
```diff
@@ -1,5 +1,57 @@
+  dt,
+  dd {
+    font-family: $font-secondary;
+    @include margin(8, "bottom");
```
Can you order @include first

Cosmin-Daniel replied:
The order has been restored!

Cosmin-Daniel commented on source/_patterns/02-organisms/content-headers/content-header-journal.mustache:
```diff
@@ -120,6 +120,10 @@
               {{> atoms-social-media-sharers-journal}}
             {{/socialMediaSharers}}
 
+            {{#definitionList}}
```
The definition list is now positioned above content-header-grid__side-popup-block__list.

viktoria-mahmud reviewed (approved):
It looks good to me!

nlisgo reviewed (changes requested):
The timeline is an entirely new block. It doesn't yet belong in pages.

@viktoria-mahmud when we discussed this we decided that it would be a new component but it could be a variant of definition-list, if appropriate. 

It certainly doesn't belong in the content header. It will be 1 of the components that is included in a new aside component.

discodavey commented on assets/sass/patterns/atoms/definition-list.scss:
```diff
@@ -1,5 +1,57 @@
 
+.definition-list {
+  @include margin(0);
+  @include margin(24, "top");
```
This can done as margin(24,0,0);

Cosmin-Daniel replied:
The margin is now added in a single row.

viktoria-mahmud commented:
> The timeline is an entirely new block. It doesn't yet belong in pages.
> 
> @viktoria-mahmud when we discussed this we decided that it would be a new component but it could be a variant of definition-list, if appropriate.
> 
> It certainly doesn't belong in the content header. It will be 1 of the components that is included in a new aside component.

viktoria-mahmud commented:
> The timeline is an entirely new block. It doesn't yet belong in pages.
> 
> @viktoria-mahmud when we discussed this we decided that it would be a new component but it could be a variant of definition-list, if appropriate.
> 
> It certainly doesn't belong in the content header. It will be 1 of the components that is included in a new aside component.

It was a temporary solution to add the element in Content header just to see how it will look like in a section but can be removed. 

I think it is suitable to have a variant of definition list, without new component, just need an example json.

Cosmin-Daniel commented:
> The timeline is an entirely new block. It doesn't yet belong in pages.
> 
> @viktoria-mahmud when we discussed this we decided that it would be a new component but it could be a variant of definition-list, if appropriate.
> 
> It certainly doesn't belong in the content header. It will be 1 of the components that is included in a new aside component.

The timeline is now removed from the content-header-journal and added as a definition-list variant.

nlisgo commented on source/_patterns/00-atoms/components/definition-list.mustache:
```diff
@@ -1,4 +1,4 @@
-<dl class="definition-list{{#inline}} definition-list--inline{{/inline}}">
+<dl class="definition-list{{#inline}} definition-list--inline{{/inline}}{{#timeline}} definition-list--timeline{{/timeline}}">
```
This should be 

```
{{#variant}} definition-list--{{variant}}{{/variant}}
```

And we can add `variant` to yaml schema as optional property:

```
  variant:
    type: string
    enum:
      - timeline
```

Cosmin-Daniel replied:
Schema is now updated with enum variant and mustache template adapted.

discodavey commented on assets/sass/patterns/atoms/definition-list.scss:
```diff
@@ -1,5 +1,56 @@
 
+.definition-list--timeline {
+  @include margin(24 0 0);
+  @include padding(24, "bottom");
```
This can be included in line 4 so not needed.

discodavey commented on assets/sass/patterns/atoms/definition-list.scss:
```diff
@@ -1,5 +1,56 @@
+    }
+
+    @media only all and (min-width: #{get-rem-from-px($bkpt-site--wide)}em) {
+      @include margin(16, "bottom");
```
Is this margin correct as it's a strange number as we normally do them in multiples of 12 but can be halved so would have thought this would have been 18px. Can we check this with Chris please?

Cosmin-Daniel replied:
I checked the design and you're right. I've updated the spaces.

nlisgo commented:
Please continue to request a review from @QueenKraken or @chris-huggins ahead of a review by myself or @discodavey

Cosmin-Daniel commented:
I created a new [ticket](https://github.com/elifesciences/issues/issues/7935) for the mobile version. We'll need to create a new component in order to achieve the design requirements, to have the content hidden/visible by dropdown.

discodavey reviewed (commented):
Looks ok to me but would get @nlisgo to check as well as he has had eyes on it more than me
